indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Elements.kt

```diff
@@ -127,6 +128,63 @@ sealed class Asn1Element(
         return result
     }
 
+
+    /**
+     * Creates a new implicitly tagged ASN.1 Element from this ASN.1 Element.
+     * NOTE: The [TagClass] of the provided [tag] will be used! If you want the result to have [TagClass.CONTEXT_SPECIFIC],
+     * also invoke `tag withClass TagClass.CONTEXT_SPECIFIC`!. If a CONSTRUCTED Tag is applied to an ASN.1 Primitive,
```

Comment:
the first point i get, the second one, not. there are exactly four legal states for those bits and no extension is possible

Reply:
the second one is purely about making it less verbose (`tag withClass CONTEXT_SPECIFIC` instead of `tag withClass TagClass.CONTEXT_SPECIFIC`)

by doing
```kotlin
sealed interface TagClass {
  companion object {
    val CONTEXT_SPECIFIC = ::CONTEXT_SPECIFIC
  }
}
data object CONTEXT_SPECIFIC: TagClass
```

Reply:
i think we should agree to disagree on this point. four values, no extensions ever possible, zero complexity in those values -> it's an enum

Reply:
my example has no added complexity 🤷 it just makes the code more readable/less verbose, same argument as something like `ULong.asImplicitTag` over `ULong withClass CONTEXT_SPECIFIC`

Reply:
neither two classes with the same name or one class which should behave like an enum, but does not is not my definition of "no added complexity", when i could just use a static import to get the same result in my code (see my last commit)